src/bundle-manifest.ts

```diff
@@ -21,15 +22,54 @@ export type UrlString = string;
  * entrypoint files.
  */
 export class Bundle {
-  public entrypoints: Set<UrlString>;
-  public files: Set<UrlString>;
+  entrypoints: Set<UrlString>;
+  files: Set<UrlString>;
+
   constructor(entrypoints?: Set<UrlString>, files?: Set<UrlString>) {
     this.entrypoints = entrypoints || new Set<UrlString>();
     this.files = files || new Set<UrlString>();
   }
 }
 
 /**
+ * A bundle manifest is a mapping of urls to bundles.
+ */
+export class BundleManifest {
+  // Map of bundle url to bundle.
+  bundles: Map<UrlString, Bundle>;
+
+  // Map of file url to bundle url.
+  files: Map<UrlString, UrlString>;
+
+  /**
+   * Given a collection of bundles and a BundleUrlMapper to generate urls for
+   * them, the constructor populates the `bundles` and `files` index properties.
+   */
+  constructor(bundles: Bundle[], urlMapper: BundleUrlMapper) {
+    const bundleUrls = urlMapper(bundles);
+
+    this.bundles = new Map();
+    this.files = new Map();
+
+    for (let i = 0; i < bundles.length; ++i) {
+      const bundle = bundles[i], bundleUrl = bundleUrls[i];
+      this.bundles.set(bundleUrl, bundle);
+      for (const fileUrl of bundle.files) {
+        this.files.set(fileUrl, bundleUrl);
```

Comment:
Good point.  In case bundle strategy or other part of bundle definition pipeline is broken.